napari/utils/transforms/transform_utils.py

```diff
@@ -466,3 +466,27 @@ def is_matrix_triangular(matrix):
     return is_matrix_upper_triangular(matrix) or is_matrix_lower_triangular(
         matrix
     )
+
+
+def is_diagonal(matrix, tol=1e-8):
+    """Determine whether affine is a diagonal matrix.
```

Comment:
```suggestion
    """Determine whether a square matrix is diagonal up to some tolerance.
```

No need to mention affine here, especially as we describe affine transforms matrices with non-zero translation to be diagonal.